dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/HttpUtils.java

```diff
@@ -467,30 +465,110 @@ public static String getProtocolHeader(Message m, String name, String defaultVal
 
     public static String getBaseAddress(Message m) {
         String endpointAddress = getEndpointAddress(m);
-        try {
-            URI uri = new URI(endpointAddress);
-            String path = uri.getRawPath();
-            String scheme = uri.getScheme();
+        //Liberty code change start
+        String[] addr = parseURI(endpointAddress, false);
+        if (addr != null) {
+            String scheme = addr[0];
+            String path = addr[1];
             if (scheme != null && !scheme.startsWith(HttpUtils.HTTP_SCHEME)
-                && HttpUtils.isHttpRequest(m)) {
+                    && HttpUtils.isHttpRequest(m)) {
                 path = HttpUtils.toAbsoluteUri(path, m).getRawPath();
             }
             return (path == null || path.length() == 0) ? "/" : path;
-        } catch (URISyntaxException ex) {
+        } else {
             return endpointAddress == null ? "/" : endpointAddress;
         }
     }
 
+    public static String[] parseURI(String addr, boolean parseAuthority) {
+        String scheme = null;
+        String parsedAuthorityOrRawPath = null;
+        int n = addr.length();
```

Comment:
I think a null check is warranted here since `getBaseAddress(...)` accounts for the possibility that `getEndpointAddress(m)` might return null.

Reply:
It actually didn't account for it returning null.  It blindly did URI uri = new URI(endpointAddress);
That call will NOT return a URISyntaxException, it will throw a NullPointerException.  Josh's new code in HttpUtils does the same thing.  If you pass in null you get NullPointerException.

Reply:
Maybe Josh should add a null test to the invalid URI test in the unit test.  In both cases should see the NPE.

Reply:
That makes sense - but in that case, there is no need for this line (previously in the catch block - not in the else block):
```
return endpointAddress == null ? "/" : endpointAddress;
```

It sounds like `endpointAddress` will never be null here - either it would've already thrown an unhandled NPE or it would be non-null.  The else block should probably be removed (no need for an else at this point anyway) and replaced with `return endpointAddress`.

Reply:
Good point.  I had seen the null check there and knew it would never happen, but didn't think it was required to make the change, but Josh can probably easily make that change.  Personally I would prefer to put the if null check and then do everything if it isn't null.